third-party/Catch2/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+therock_subproject_fetch(therock-catch2-sources
+  CMAKE_PROJECT
+  URL https://github.com/catchorg/Catch2/archive/refs/tags/v3.7.0.tar.gz
```

Comment:
Because this is v3, I'm assuming this builds an actual library vs header only. We probably want to make sure that is built as static vs dynamic (that is how we set up gtest).

Also, if used in shared libraries, we need to make sure to force pic mode on. Can avoid that if not, but there are a few cases where gtest is used that way. 

If those conditions are met, it can just be a build dep vs runtime dep in consumers.